### PR TITLE
Refactor ToIP on DialArgs

### LINKED ISSUE #68: ToIP doesn't work with QUIC/WS/etc. addresses
Currently, it uses ToNetAddr. However, apparently, ToNetAddr will only return TCP/UDP addresses for actual TCP/UDP multiaddrs, not multiaddrs that simply start with TCP/UDP. That is:

* `/ip4/.../tcp/...` will work.
* `/ip4/.../tcp/.../ws` won't.

Ideally, we'd fix https://github.com/multiformats/go-multiaddr-net/issues/67 then just fix `ToIP` by using that new function.

### PR DESCRIPTION
**Pull request by willscott**
address #67
fix #68

### CODE REVIEW

File: convert.go [hunk 1]
```diff
@@ -100,20 +100,23 @@ func FromIP(ip net.IP) (ma.Multiaddr, error) {
...
 
-	switch netAddr := n.(type) {
-	case *net.UDPAddr:
-		return netAddr.IP, nil
-	case *net.TCPAddr:
-		return netAddr.IP, nil
-	case *net.IPAddr:
-		return netAddr.IP, nil
+	if hostname {
+		addr, err := net.ResolveIPAddr(network, ip)
```
Comment (Stebalien) [inline convert.go:109]: Hm. I didn't even _realize_ we were resolving these. We probably shouldn't do that. I wouldn't expect `ToIP(...)` to make a network request!

Honestly, I'd just reject this case.

Comment (hsanjuan) [reply convert.go:109]: This was not happening I think?

Comment (willscott) [reply convert.go:109]: correct. DialArgs hasn't resolved and returns the string hostname. `ToNetAddr` may do lookups, as it delegates to `ResolveTCPAddr` in the basic codecs, so asking for a NetAddr of a malformed multiaddr like `/tcp4/google.com` would do a DNS resolution for google.

Comment (Stebalien) [reply convert.go]: Encoding that multiaddr should fail.

Comment (willscott) [reply convert.go]: In latest commit it is changed to fail

**Pull request merged**